modules/cudacodec/src/video_reader.cpp

```diff
@@ -191,9 +150,17 @@ namespace
         videoSource_->setVideoParser(videoParser_);
         videoSource_->start();
         waitForDecoderInit();
+        FormatInfo format = videoDecoder_->format();
+        if (format.colorSpaceStandard == ColorSpaceStandard::Unspecified) {
+            if (format.width > 1280 || format.height > 720)
+                format.colorSpaceStandard = ColorSpaceStandard::BT709;
+            else
+                format.colorSpaceStandard = ColorSpaceStandard::BT601;
+        }
```

Comment:
Is it some heuristic? Do you have any reference for the decision?

Reply:
Yes, from my understanding 709 is for HD but I could't find a definitive source to say if there was an exact definition of what resolution this starts at.  Although many sources mentioned 720p as the minimum resolution in the end I chose 720p as it replicated the color format behaviour from sources decoded with the `VideoCapture` class..